src/Nethermind/Nethermind.Network/PeerManager.cs

```diff
@@ -288,9 +289,17 @@ private async Task RunPeerUpdateLoop()
                         workerBlock.Complete();
 
                         // Wait for all messages to propagate through the network.
-                        workerBlock.Completion.Wait();
+                        await workerBlock.Completion;
 
                         Interlocked.Increment(ref _connectionRounds);
+
+                        long nowMs = Environment.TickCount64;
+                        long diffMs = nowMs - lastMs;
+                        if (diffMs < 50)
+                        {
+                            await Task.Delay(50 - (int)diffMs);
+                        }
+                        lastMs = nowMs;
```

Comment:
I would say go to periodic timer :D
Also is 50ms not too small? Should we wait more?

Reply:
Is an outer main timer set on `_networkConfig.PeersUpdateInterval` probably could be a periodic timer ;)

This is the inner loop where its running through the 100k+ stored peers at (Max - connected) at a time saying "hey, you there?"

Reply:
Actually probably right; after going through them all, should wait a little additional extra before starting again https://github.com/NethermindEth/nethermind/pull/5751